src/OrchardCore.Modules/OrchardCore.Users/Drivers/UserInformationDisplayDriver.cs

```diff
@@ -14,73 +16,103 @@ public class UserInformationDisplayDriver : DisplayDriver<User>
+            // validate the user and provide the correct error messages based on the entire user objects values.
 
-                // Custom properties should still be validated in the driver.
+            // Custom properties should still be validated in the driver.
 
-                user.UserName = model.UserName;
-                user.Email = model.Email;
+            if (context.IsNew)
+            {
+                if (await context.Updater.TryUpdateModelAsync(userNameModel, Prefix))
+                {
+                    user.UserName = userNameModel.UserName;
+                }
+
+                if (await context.Updater.TryUpdateModelAsync(emailModel, Prefix))
+                {
+                    user.Email = emailModel.Email;
+                }
             }
+            else
+            {
+                var site = await _siteService.GetSiteSettingsAsync();
+                var settings = site.As<LoginSettings>();
 
-            return Edit(user);
-        }
+                if (settings.AllowChangingUsername && await CanEditUserInfoAsync(user) && await context.Updater.TryUpdateModelAsync(userNameModel, Prefix))
+                {
+                    user.UserName = userNameModel.UserName;
+                }
 
-        private Task<bool> AuthorizeUpdateAsync(User user)
-        {
-            // When the current user matches this user we can ask for ManageOwnUserInformation
-            if (String.Equals(user.UserId, _httpContextAccessor.HttpContext.User.FindFirstValue(ClaimTypes.NameIdentifier), StringComparison.OrdinalIgnoreCase))
-            {
-                return _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, Permissions.ManageOwnUserInformation);
+                if (settings.AllowChangingEmail && await CanEditUserInfoAsync(user) && await context.Updater.TryUpdateModelAsync(emailModel, Prefix))
+                {
+                    user.Email = emailModel.Email;
+                }
             }
 
-            // Otherwise we require permission to manage this users information.
-            return _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, Permissions.ManageUsers, user);
+            return await EditAsync(user, context);
         }
 
-        private Task<bool> AuthorizeEditAsync(User user)
+        private async Task<bool> CanEditUserInfoAsync(User user)
         {
-            // When the current user matches this user we can ask for ManageOwnUserInformation
-            if (String.Equals(user.UserId, _httpContextAccessor.HttpContext.User.FindFirstValue(ClaimTypes.NameIdentifier), StringComparison.OrdinalIgnoreCase))
-            {
-                return _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, Permissions.ManageOwnUserInformation);
-            }
+            return !IsCurrentUser(user) || await _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, Permissions.ManageOwnUserInformation);
```

Comment:
Yes for info I had the following use case.

I could create an user in a given Role1 that can edit users in Role1 and another given role2. Then this user was able to edit users in Role2 and change name/email (allowed in loginSettings), but was not able to edit and then do the same for itself even it is in Role1, this because this user doesn't have `ManageOwnUserInformation`, maybe intended.

But was the same even if I added to this role the `EditUsers` permission, this I think because, even if we check `EditUsers` at the beginning, here we still check `ManageOwnUserInformation` which currently is not implied by `EditUsers`. I will comment it in the related permissions file.

Reply:
@jtkech  I am having hard time following this comment. Are you saying these is an issue? If so, can you explain it in steps so I can reproduce it and fix it? Or, you are more than welcome to fix it directly if it is easier.

Reply:
Resolved as ManageOwnUserInformation is now implied by EditUsers.